mitiq/factories.py

```diff
@@ -0,0 +1,130 @@
+from pyquil import Program
+from typing import List, Tuple, Callable
+
+import mitiq.qiskit.qiskit_utils as qs_utils
+import numpy as np
+
+class Factory(object):
+    """
+    This object adaptively produces new noise scaling parameters based on a historical stack of previous noise
+    scale parameters and previously calculated expectation values. Noise scaling parameters are stored on the `in_stack`
+    and the running list of expectation values are stored on the `out_stack`.
+    """
+    def __init__(self, instack: List[float], outstack: List[float]) -> None:
+        self.instack = instack
+        self.outstack = outstack
+
+    def step(self, instack: List[float], outstack: List[float]) -> None:
+        raise NotImplementedError
+
+    def is_converged(self, instack: List[float], outstack: List[float]) -> None:
+        raise NotImplementedError
+
+    def reduce(self) -> float:
+        raise NotImplementedError
+
+
+class BatchedFactory(Factory):
+    def __init__(self, scalars: List[float], instack: List[float] = None, outstack: List[float] = None) -> None:
```

Comment:
Change type of `scalars` to be `Iterable[float]`. It's natural to do something like `scalars = np.arange(start, stop, step)`, so we should account for numpy arrays or any other iterable object of floats.